internal/querynodev2/segments/segment.go

```diff
@@ -356,13 +356,59 @@ func NewSegmentV2(
 		insertCount:        atomic.NewInt64(0),
 	}
 
-	if segmentType != SegmentTypeSealed {
-		segment.loadStatus.Store(string(LoadStatusInMemory))
+	if err := segment.initializeSegment(); err != nil {
+		return nil, err
 	}
-
 	return segment, nil
 }
 
+func (s *LocalSegment) initializeSegment() error {
+	loadInfo := s.loadInfo.Load()
```

Comment:
initialize all go side field at create segment but not in segment loader (may be call multiple time if set it in segment_loader).